packages/plugins/users-permissions/server/strategies/users-permissions.js

```diff
@@ -67,53 +67,26 @@ const authenticate = async ctx => {
 const verify = async (auth, config) => {
   const { credentials: user } = auth;
 
-  // public accesss
-  if (!user) {
-    // test against public role
-    const publicPermissions = await strapi.query('plugin::users-permissions.permission').findMany({
-      where: {
-        role: { type: 'public' },
-      },
-    });
-
-    const allowedActions = map('action', publicPermissions);
-
-    // A non authenticated user cannot access routes that do not have a scope
-    if (!config.scope) {
-      throw new UnauthorizedError();
-    }
-
-    const isAllowed = castArray(config.scope).every(scope => allowedActions.includes(scope));
-
-    if (!isAllowed) {
-      throw new ForbiddenError();
-    }
-
-    return;
+  // A non authenticated user cannot access routes that do not have a scope
+  if (!user && !config.scope) {
+    throw new UnauthorizedError();
   }
 
-  const permissions = await strapi.query('plugin::users-permissions.permission').findMany({
-    where: { role: user.role.id },
-  });
+  let allowedActions = auth.allowedActions;
 
-  const allowedActions = map('action', permissions);
+  if (!allowedActions) {
+    const permissions = await strapi.query('plugin::users-permissions.permission').findMany({
+      where: { role: user ? user.role.id : { type: 'public' } },
+    });
 
-  // An authenticated user can access non scoped routes
-  if (!config.scope) {
-    return;
+    allowedActions = auth.allowedActions = map('action', permissions);
```

Comment:
can you split this into multiple assignments instead

Reply:
I did not quite understand what you want, but I noticed that I missed one more important condition

Reply:
can you avoid using = twice on the same line but do two assignements :)